nixos/modules/services/monitoring/prometheus/exporters/script.nix

```diff
@@ -0,0 +1,46 @@
+{ config, lib, pkgs, options }:
+
+with lib;
+
+let
+  cfg = config.services.prometheus.exporters.script;
+  configFile = pkgs.writeText "script-exporter.yaml" (builtins.toJSON cfg.settings);
+in
+{
+  port = 9172;
+  extraOpts = {
+    settings = mkOption {
+      type = types.attrs;
+      default = {};
```

Comment:
Does it make sense to run the exporter without any scripts defined? If not, I'd leave the `default` out.